morango/sync/context.py

```diff
@@ -51,8 +51,9 @@ def __init__(
 
         if self.transfer_session:
             self.sync_session = transfer_session.sync_session or self.sync_session
-            self.filter = transfer_session.get_filter() or self.filter
             self.is_push = transfer_session.push or self.is_push
+            if transfer_session.filter and transfer_session.filter is not None:
```

Comment:
If it's `None`, wouldn't it have already failed the falsiness check in the first half of the `and`? So the second half may not be necessary?